base.go

```diff
@@ -36,6 +37,75 @@ func GetBaseLayer(o BaseLayerOpts) error {
 	}
 }
 
+func getOci(o BaseLayerOpts) error {
+	url, err := o.Layer.From.ParseTag()
+	if err != nil {
+		return err
+	}
+	// url is "ocidir:image"
+	res := strings.SplitN(url, ":", 2)
+	fullImage := res[1]
+	// image may be "imagebase_tag" i.e. boot_0.1.3
+	res = strings.SplitN(fullImage, "_", 2)
+	imageBase := fullImage
+	tag := ""
+	if len(res) == 2 {
+		imageBase = res[0]
+		tag = res[1]
+	}
+
+	cacheDir := path.Join(o.Config.StackerDir, "layer-bases", tag)
+	if err := os.MkdirAll(cacheDir, 0755); err != nil {
+		return err
+	}
+
+	cmd := exec.Command(
+		"skopeo",
+		"copy",
+		fmt.Sprintf("oci:%s",url),
+		fmt.Sprintf("oci:%s:%s", cacheDir, imageBase),
+	)
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("skopeo copy: %s", err)
+	}
+
+	// We just copied it to the cache, now let's copy that over to our image.
```

Comment:
Any chance we can abstract this since it's copy pasted from something else?

Reply:
Yeah I'm going to close this PR - I think the abstraction needs to be fixed up.  The 'url' is not generic enough.  We want a type struct that has a generic 'source' struct (which can be a dockerhub url or an oci layout directory), an image name, and maybe optional additional tag.